CLIENT-3931 CLIENT-3932

DomPeliniAerospike opened the pull request:
…ded to record if possible.

DomPeliniAerospike committed:
Fixed bin name length check, added checking to ensure all bins are added to record if possible.

jefflembeck commented:
Worth thinking about, since this will hit the typescript code, maybe adding some jsdoc to the type for bin names?

Currently we have:

```typescript
/**
 * Represents an object containing one or more `AerospikeBinValues` with associated string keys.
 */
export type AerospikeBins = {
    [key: string]: any
};
```

it might be nice to have:

```typescript
/**
 * Bin name - maximum 15 characters.
 * @remarks Names exceeding 15 characters will result in ERR_PARAM at runtime.
 */
type BinName = string;

/**
 * Represents an object containing one or more `AerospikeBinValues` with associated string keys.
 */
export type AerospikeBins = {
    [key: BinName]: any
};
```

as an addition. Just for the sheer editor help.

(with hope for changing to

```typescript
export type AerospikeBins = {
    [key: BinName]: AerospikeBinValue
};
```

eventually)

DomPeliniAerospike committed:
Added typescript fixes for AerospikeRecord and AerospikeBins.  Contributor: @jefflembeck

DomPeliniAerospike commented:
Just added a new commit fixing the following

[BatchResult.record incorrectly typed as B instead of AerospikeRecord<B>](https://github.com/aerospike/aerospike-client-nodejs/issues/802)
[AerospikeRecord.bins incorrectly typed as AerospikeRecord<B> instead of B](https://github.com/aerospike/aerospike-client-nodejs/issues/801).
Added binName type and added to AerospikeBins.
Changed any to AerospikeBinValue in AerospikeBins.

DomPeliniAerospike committed:
Merge branch 'dev' into CLIENT-3931-CLIENT-3932
